Sources/Managers/ApiManager.swift

```diff
@@ -18,43 +20,57 @@ class ApiManager {
         }
     }
 
-    func getUrlRequest(searchTerm: String) throws -> URLRequest? {
-        var urlComponents = URLComponents()
-        urlComponents.scheme = "https"
-        urlComponents.host = "api.github.com"
-        urlComponents.path = "/search/repositories"
-        urlComponents.queryItems = [
-            URLQueryItem(name: "q", value: searchTerm)
-        ]
-
-        guard let url = urlComponents.url else {
-            throw ApiError.invalidURL
-        }
-
-        return URLRequest(url: url)
-    }
-
     var decoder: JSONDecoder {
         let decoder = JSONDecoder()
         decoder.keyDecodingStrategy = .convertFromSnakeCase
         decoder.dateDecodingStrategy = .iso8601
         return decoder
     }
 
-    func getRepositories(for search: String) async throws -> [Repository] {
-        guard let urlRequest = try getUrlRequest(searchTerm: search)
+    func fetch<T: Codable>(
+        type: T.Type,
+        endpoint: Endpoint
+    ) async throws -> T {
+        let (data, response) = try await URLSession.shared
+            .data(for: endpoint.request)
+        guard (response as? HTTPURLResponse)?.statusCode == 200
         else {
             throw ApiError.badURLRequest
         }
-        let (data, response) = try await URLSession.shared.data(for: urlRequest)
-
-        guard (response as? HTTPURLResponse)?.statusCode == 200
-        else {
-            print("An error occured with status: \(response)")
-            return []
+        do {
+            return try ApiManager.shared.decoder.decode(type, from: data)
+        } catch {
+            print(error)
+            throw ApiError.dataNotFound(description: error.localizedDescription)
         }
+    }
+
+    enum Endpoint {
+        case users(user: String)
+        case searchRepositories(query: String)
+        case userRepository(userLogin: String)
```

Comment:
since in the users enum case you are using _user_ as a parameter name, you can change userRepository parameter to _user_ as well instead of _userLogin_

also, userRepository seems wrong, if you are fetching every repository of a particular user, you probably want to rename it to something like _**userRepositories**_
also, users is fetching single user details so _**user**_ seems more convenient